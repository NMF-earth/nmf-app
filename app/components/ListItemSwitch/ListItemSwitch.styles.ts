import { StyleSheet } from "react-native";

import { Colors } from "style";

export default StyleSheet.create({
  container: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    flex: 1,
  },
  text: {
    paddingVertical: 18,
  },
  topLine: {
    borderTopColor: Colors.grey10,
    borderTopWidth: 1.6,
  },
  bottomLine: {
    borderBottomColor: Colors.grey10,
    borderBottomWidth: 1.6,
  },
});
