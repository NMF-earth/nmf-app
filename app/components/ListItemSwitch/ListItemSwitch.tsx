import React from "react";
import { View, ViewStyle, StyleProp } from "react-native";
import { Switch } from "react-native-gesture-handler";

import styles from "./ListItemSwitch.styles";
import Text from "../Text";

interface Props {
  title: string;
  value: boolean;
  onChange: () => void;
  showTopLine?: boolean;
  showBottomLine?: boolean;
}

const ListItemSwitch: React.FC<Props> = ({
  title,
  value,
  onChange,
  showTopLine,
  showBottomLine,
}) => {
  const containerStyle: StyleProp<ViewStyle> = [styles.container];
  if (showTopLine) {
    containerStyle.push(styles.topLine);
  }

  if (showBottomLine) {
    containerStyle.push(styles.bottomLine);
  }

  return (
    <View style={containerStyle}>
      <Text.Secondary style={styles.text}>{title}</Text.Secondary>
      <Switch value={value} onChange={onChange} />
    </View>
  );
};

export default ListItemSwitch;
