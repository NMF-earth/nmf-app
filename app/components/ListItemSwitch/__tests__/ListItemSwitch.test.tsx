import "react-native";
import React from "react";
import { create } from "react-test-renderer";

import ListItemSwitch from "../ListItemSwitch";

describe("<ListItemSwitch />", () => {
  const props = {
    title: "Germany",
    value: true,
    onChange: () => {
      // do nothing.
    },
  };
  const wrapper = create(<ListItemSwitch {...props} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("ListItem renders correctly with topLine", () => {
    const tree = create(<ListItemSwitch showTopLine {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("ListItem renders correctly with bottomLine", () => {
    const tree = create(<ListItemSwitch showBottomLine {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("ListItem renders correctly with both topLine and bottomLine", () => {
    const tree = create(<ListItemSwitch showBottomLine showTopLine {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
