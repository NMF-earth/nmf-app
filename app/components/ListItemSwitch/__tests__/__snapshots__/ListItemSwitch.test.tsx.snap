// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItemSwitch /> ListItem renders correctly with both topLine and bottomLine 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      {
        "borderTopColor": "#F0F0F0",
        "borderTopWidth": 1.6,
      },
      {
        "borderBottomColor": "#F0F0F0",
        "borderBottomWidth": 1.6,
      },
    ]
  }
>
  <Text.Secondary
    style={
      {
        "paddingVertical": 18,
      }
    }
  >
    Germany
  </Text.Secondary>
  <RCTSwitch
    accessibilityRole="switch"
    collapsable={false}
    handlerTag={4}
    handlerType="NativeViewGestureHandler"
    onChange={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "height": 31,
        "width": 51,
      }
    }
    value={true}
  />
</View>
`;

exports[`<ListItemSwitch /> ListItem renders correctly with bottomLine 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      {
        "borderBottomColor": "#F0F0F0",
        "borderBottomWidth": 1.6,
      },
    ]
  }
>
  <Text.Secondary
    style={
      {
        "paddingVertical": 18,
      }
    }
  >
    Germany
  </Text.Secondary>
  <RCTSwitch
    accessibilityRole="switch"
    collapsable={false}
    handlerTag={3}
    handlerType="NativeViewGestureHandler"
    onChange={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "height": 31,
        "width": 51,
      }
    }
    value={true}
  />
</View>
`;

exports[`<ListItemSwitch /> ListItem renders correctly with topLine 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
      {
        "borderTopColor": "#F0F0F0",
        "borderTopWidth": 1.6,
      },
    ]
  }
>
  <Text.Secondary
    style={
      {
        "paddingVertical": 18,
      }
    }
  >
    Germany
  </Text.Secondary>
  <RCTSwitch
    accessibilityRole="switch"
    collapsable={false}
    handlerTag={2}
    handlerType="NativeViewGestureHandler"
    onChange={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "height": 31,
        "width": 51,
      }
    }
    value={true}
  />
</View>
`;

exports[`<ListItemSwitch /> render 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      },
    ]
  }
>
  <Text.Secondary
    style={
      {
        "paddingVertical": 18,
      }
    }
  >
    Germany
  </Text.Secondary>
  <RCTSwitch
    accessibilityRole="switch"
    collapsable={false}
    handlerTag={1}
    handlerType="NativeViewGestureHandler"
    onChange={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "height": 31,
        "width": 51,
      }
    }
    value={true}
  />
</View>
`;
